abdiff/core/exceptions.py

```diff
@@ -3,30 +3,25 @@ def __init__(self, run_id: str) -> None:
         super().__init__(f"No Docker containers were found with label run='{run_id}'.")
 
 
-class DockerContainerRuntimeExceededTimeoutError(Exception):
-    def __init__(self, containers: list, timeout: int) -> None:
-        self.containers = containers
+class DockerContainerTimeoutError(Exception):
+    def __init__(self, container_id: str | None, timeout: int) -> None:
+        self.container_id = container_id
         self.timeout = timeout
         super().__init__(self.get_formatted_message())
 
     def get_formatted_message(self) -> str:
-        container_ids = [container.id for container in self.containers]
-        return (
-            f"Timeout of {self.timeout} seconds exceeded."
-            f"{len(container_ids)} container(s) is/are still running:"
-            f"{container_ids}."
-        )
+        return f"Container {self.container_id} exceed timeout of {self.timeout} seconds."
 
 
 # core function errors
-class DockerContainerRunFailedError(Exception):
-    def __init__(self, containers: list) -> None:
-        self.containers = containers
+class DockerContainerRuntimeError(Exception):
```

Comment:
A bit of exception name renaming -- apologies for not raising on previous PRs -- to keep the names a bit shorter and more standardized.

Reply:
I think one of the reasons I went with the original name was realizing that _if_ we ran the containers with `detach=False`, the [`docker.client.containers.run` command](https://docker-py.readthedocs.io/en/stable/containers.html#container-objects) would raise the following errors: 
<img width="709" alt="image" src="https://github.com/user-attachments/assets/eac519e2-0715-461d-bf67-8478ac2aeda1">.

@ghukill Wondering if you have any ideas on how to convince (lolz) the module's error to bubble up (maybe I missed something in the documentation)? In any case, I agree that the rename of the custom exception is good. 👍🏼

Reply:
I'm not sure if I'm totally answering your question @jonavellecuerdo, but yesterday and this comment got me thinking that perhaps we should _not_ launch the containers in a detached state?  

I did so in this work to keep the majority of the code as-was; where we get a `Container` object back and can pass that around.

I was considering exploring this option, but I feel like it's ultimately safer to launch in a detatched state, which allows us to monitor it's execution and have more control of stopping it when needed.  I'm concerned that if we launch a thread, and then invoke `docker_client.containers.run(..., detach=False)` from within that thread, and then _kill_ the thread because we determined it has timed out, that it will _not_ stop the container.  It might... but it might not.  

Ultimately, I quite like the pattern you implemented @jonavellecuerdo with the `detach=True`, and then we have a `Container` object to work with and pass around.

As for these errors... I think it's unlikely that we can have them bubble up.  But I think that's okay at this time.